pkg/ds/daemon_set.go

```diff
@@ -0,0 +1,347 @@
+package ds
+
+import (
+	"time"
+
+	"github.com/square/p2/pkg/util"
+
+	klabels "k8s.io/kubernetes/pkg/labels"
+
+	"github.com/square/p2/pkg/ds/fields"
+	"github.com/square/p2/pkg/kp"
+	"github.com/square/p2/pkg/kp/dsstore"
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/scheduler"
+	"github.com/square/p2/pkg/types"
+)
+
+const (
+	// This label is applied to pods owned by an RC.
+	DSIDLabel = "daemon_set_id"
+)
+
+type DaemonSet interface {
+	ID() fields.ID
+
+	WatchDesires(
+		quitCh <-chan struct{},
+		updatedCh <-chan fields.DaemonSet,
+		deletedCh <-chan struct{},
+		nodesChangedCh <-chan struct{},
+	) <-chan error
+
+	// CurrentPods() returns all nodes that are scheduled by this daemon set
+	CurrentPods() (PodLocations, error)
+}
+
+type PodLocation struct {
```

Comment:
Should we deduplicate this type and the `Nodes` function bweteen the `rc` package and this one by moving it somewhere?